### PR TITLE
Bonnie 976 profile

### PR DESCRIPTION
**Pull request by alexanderelliott121**
JIRA: https://jira.mitre.org/browse/BONNIE-976
Test: https://jira.mitre.org/browse/BONNIE-1030

### CODE REVIEW

**Comment by alexanderelliott121** [comment]
In Gemfile, point health-data-standards at BONNIE-976_Profile

**Review by c-monkey — APPROVED** [review]
Unit tests here, in HDS, and in bonnie pass.

https://jira.mitre.org/browse/BONNIE-1030 passed.

**Review by edeyoung — DISMISSED** [review]
Several conflicts with base branch (likely due to the huge number of value-set related changes I made).

Need to make sure that client-side valueset retrieves appropriately get the correct value set. Currently, I don't believe it will break because `version` is null so it just grabs the latest value set. However, it should actually grab the value set with the correct version. This isn't just for processing, but also for populating the data criteria as well as the value set list. See where the value set changes touched things in the recent value set change commit.

**Review by c-monkey — APPROVED** [review]
the unit tests in bonnie, bonnie_bundler, and hds pass; https://jira.mitre.org/browse/BONNIE-1030 passes

**Review by jbradl11 — APPROVED** [review]
Tests pass. Rebuild elm works as expected.

**Comment by lukoktonos** [comment]
Regression script passed--all results are the same.

File: lib/measures/loading/value_set_loader.rb [hunk 1]
```diff
@@ -142,9 +142,15 @@ def self.load_value_sets_from_vsac(value_sets, username, password, user=nil, ove
         RestClient.proxy = ENV["http_proxy"]
         value_sets.each do |value_set|
           value_set_version = value_set[:version] ? value_set[:version] : "N/A"
+          value_set_profile = (value_set[:profile] && !includeDraft) ? value_set[:profile] : nlm_config["profile"]
+          vs_query_params = {user_id: user.id, oid: value_set[:oid]}
+          if value_set[:profile]
```
Comment (jbradl11) [inline lib/measures/loading/value_set_loader.rb]: Could you provide a comment here about what is going on and the reason why you are assigning `value_set_profile` to `vs_query_params[:version]` ? I know whats going on but it might be useful to document it for the future.

File: lib/measures/loading/value_set_loader.rb [hunk 2]
```diff
@@ -162,15 +168,16 @@ def self.load_value_sets_from_vsac(value_sets, username, password, user=nil, ove
                 # this in our rebuild elm rake task so that we retrieve the appropriate value sets.
                 # It is currently not an issue because we ignore include_draft when a version is specified and when a version is not specified,
                 # it is set to "N/A" and continues to be referenced in that way.
-                vs_data = api.get_valueset(value_set[:oid], version: value_set[:version], effective_date: effectiveDate, include_draft: false, profile: nlm_config["profile"])
+                vs_data = api.get_valueset(value_set[:oid], version: value_set[:version], include_draft: false, profile: nlm_config["profile"])
```
Comment (jbradl11) [inline lib/measures/loading/value_set_loader.rb]: Is the comment above still relevant?  Or should the comment be expanded?

Comment (alexanderelliott121) [reply lib/measures/loading/value_set_loader.rb]: Added Details

File: lib/measures/loading/value_set_loader.rb [hunk 3]
```diff
@@ -197,7 +202,7 @@ def self.load_value_sets_from_vsac(value_sets, username, password, user=nil, ove
               # As of t9/7/2017, when valuesets are retrieved from VSAC via profile, their version defaults to N/A
               # As such, we set the version to the profile with an indicator.
               if value_set[:profile]
-                set.version = "Profile:#{value_set_profile}"
+                set.version = value_set_profile
```
Comment (jbradl11) [inline lib/measures/loading/value_set_loader.rb]: Why was this changed? What was the issue it was causing?

**Comment by jbradl11** [comment]
I am getting multiple failures when running the unit tests.

File: lib/measures/loading/value_set_loader.rb (hunk 3 above)
Comment (edeyoung) [reply lib/measures/loading/value_set_loader.rb]: yes, i also have this question - we did the "profile" because some profiles are the same as some versions and we didn't want to cause confusion - we wanted to make sure they were differentiated.

**Comment by holmesie** [comment]
I am getting a test error when running the unit tests:
```
  1) Error:
LoadSourcesTest#test_Loading_measure_from_sources:
Measures::VSACException: Error Loading Value Sets from VSAC: Error Loading Value Sets from VSAC: undefined method `id' for nil:NilClass
```

**Review by holmesie — APPROVED** [review]
Tests pass, looks good

**Review by jbradl11 — APPROVED** [review]
All code looks good. Jira Test passes and unit tests pass.

**Comment by edeyoung** [comment]
travis builds are just frustratingly breaking now. this is a problem for post 9/12. all tests pass locally across several machines.
